services/wallet/saved_addresses.go

```diff
@@ -82,8 +83,18 @@ func (sam *SavedAddressesManager) GetRawSavedAddresses() ([]SavedAddress, error)
 }
 
 func (sam *SavedAddressesManager) upsertSavedAddress(sa SavedAddress, tx *sql.Tx) error {
-	sqlStatement := "INSERT OR REPLACE INTO saved_addresses (address, name, favourite, removed, update_clock, chain_short_names, ens_name, is_test) VALUES (?, ?, ?, ?, ?, ?, ?, ?)"
-	var err error
+	savedAddresses, err := sam.GetSavedAddresses()
+	if err != nil {
+		return err
+	}
+	sa.CreatedAt = time.Now().Unix()
+	for _, savedAddress := range savedAddresses {
+		if savedAddress.Address == sa.Address && savedAddress.IsTest == sa.IsTest && savedAddress.ENSName == sa.ENSName {
+			sa.CreatedAt = savedAddress.CreatedAt
```

Comment:
This approach will also update the `created_at` value for updated addresses. Is this intended? Then the column name should be `updated_at`.

As others noted, this should be done in SQL.
The simplest way to do it, for the update case done here, is to change the query with:
```sql
INSERT OR REPLACE INTO saved_addresses (address, name, favourite, removed, update_clock, chain_short_names, ens_name, is_test, updated_at) VALUES (?, ?, ?, ?, ?, ?, ?, ?, strftime('%s', 'now'))
```
See https://www.sqlite.org/lang_datefunc.html

Reply:
Sorry for the confusion. I meant to say that the entire implementation here should be replaced with the SQL query change I mentioned and still get the same result.

Reply:
I missed the identity condition up there, thanks @alaibe and @IvanBelyakoff for raising this.